pkg/plugins/manager/loader/angular/angulardetector/angulardetector.go

```diff
@@ -0,0 +1,69 @@
+package angulardetector
+
+import (
+	"bytes"
+	"context"
+	"regexp"
+)
+
+var (
+	_ Detector = &ContainsBytesDetector{}
+	_ Detector = &RegexDetector{}
+
+	_ DetectorsProvider = &StaticDetectorsProvider{}
+	_ DetectorsProvider = SequenceDetectorsProvider{}
+)
+
+// Detector implements a check to see if a js file is using angular APIs.
+type Detector interface {
+	// Detect takes the content of a js file and returns true if the plugin is using Angular.
+	Detect(js []byte) bool
```

Comment:
nit: this is later used as `detector.Detect(...)` which doesn't say what we are detecting. I suggest to change it to `DetectAngularPatterns` or something like that.